cannon/mipsevm/memory/memory.go

```diff
@@ -45,141 +44,62 @@ var zeroHashes = func() [256][32]byte {
 }()
 
 type Memory struct {
-	// generalized index -> merkle root or nil if invalidated
-	nodes map[uint64]*[32]byte
-
-	// pageIndex -> cached page
-	pages map[Word]*CachedPage
-
-	// Note: since we don't de-alloc pages, we don't do ref-counting.
+	merkleIndex PageIndex
+	// Note: since we don't de-alloc Pages, we don't do ref-counting.
 	// Once a page exists, it doesn't leave memory
+	pageTable map[Word]*CachedPage
```

Comment:
Can we update this comment to indicate that this map instance may be shared with the PageIndex implementation.